RocketChat/Rocket.Chat.ReactNative
# Chore: Migrate react-navigation to TypeScript

## Proposed changes

Here we start the migration to TypeScript for our navigations' stack and drawer. In addition to this, we start as well working with the global definitions for TypeScript, which is going to help us prevent repeating interfaces across multiple screens and containers, such as `IAttachment`, `IRoom` and so on.

## Issue(s)	


## How to test or reproduce

The whole navigation across the app was changed, so I do recommend testing every existing screen on the app. Even though I mentioned that we should test the whole app, there are some particulars that we should check in case there was any possible regression, either visually or functionally:

1. Check for visual regressions on `Modals` for tablets:
    1. Go to a room
    2. Go to Room Actions 
2. Check for functional regressions on `AddChannelTeamView`
    1. Create a team
    2. Go to team's channels
    3. Go to add channel
    4. Create a channel or add an existing one
3. Check for functional regressions on the emoji picker

## Screenshots

## Types of changes




## Checklist



## Known issues

It seems like the stack folder is ignored by the `tsc` so I could commit without it checking the errors my code has. This is a known issue and might get solved in a future PR.
